scripts/spack/configs/toss_4_x86_64_ib_cray/spack.yaml

```diff
@@ -22,15 +22,35 @@ spack:
       environment: {}
       extra_rpaths: []
       flags:
+        # -Wno-int-conversion flag needed for building HDF5
+        cflags: -Wno-int-conversion
         fflags: -Mfreeform
       modules: []
       operating_system: rhel8
       paths:
-        cc: /opt/rocm-5.2.3/llvm/bin/amdclang
-        cxx: /opt/rocm-5.2.3/llvm/bin/amdclang++
-        f77: /opt/rocm-5.2.3/llvm/bin/amdflang
-        fc: /opt/rocm-5.2.3/llvm/bin/amdflang
-      spec: clang@14.0.0
+        cc: /opt/rocm-5.4.3/llvm/bin/amdclang
+        cxx: /opt/rocm-5.4.3/llvm/bin/amdclang++
+        f77: /opt/rocm-5.4.3/llvm/bin/amdflang
+        fc: /opt/rocm-5.4.3/llvm/bin/amdflang
+      spec: clang@15.0.0
+      target: x86_64
+  - compiler:
+      environment: {}
+      extra_rpaths: []
+      # -Wno-int-conversion flag needed for building HDF5
+      # -O1 flag necessary for bypassing HIP out of memory error
+      flags:
+        cflags: -Wno-int-conversion -O1
+        cxxflags: -O1
```

Comment:
Optimization flag necessary to bypass HIP `error: local memory (XXX) exceeds limit (65536)` with default optimization flag `-O0`.
Error is documented on [El Capitan JIRA](https://rzlc.llnl.gov/jira/browse/ELCAP-223)